test/ng/qSpec.js

```diff
@@ -2167,45 +2165,69 @@ describe('q', function() {
 
 
   describe('when exceptionHandler is called', function() {
-    it('should log an unhandled rejected promise', function() {
-      var defer = q.defer();
-      defer.reject('foo');
-      mockNextTick.flush();
-      expect(exceptionHandlerStr()).toBe('Possibly unhandled rejection: foo');
-    });
+    var exceptionEg = new Error('Fail');
+    var exceptionStr = toDebugString(exceptionEg);
+    var fixtures = [
+      {
+        type: 'exception',
+        value: exceptionEg,
+        expected: {
+          exception: exceptionEg,
+          reason: 'Possibly unhandled rejection: ' + exceptionStr
+        }
+      },
+      {
+        type: 'plain value',
+        value: 'foo',
+        expected: {
+          reason: 'Possibly unhandled rejection: foo'
+        }
+      }
```

Comment:
A couple more fixtures would be in order imo. E.g.: 

- One with a non-`Error` object.
- One with an object that "extends" `Error`.